src/libutil/typedesc.cpp

```diff
@@ -175,8 +175,6 @@ TypeDesc::c_str () const
     std::string result;
     if (aggregate == SCALAR)
         result = basetype_name[basetype];
-    else if (aggregate == MATRIX44 && basetype == FLOAT)
-        result = "matrix";
```

Comment:
Yeah, I left the `fromstring` method so that "matrix" still converts to `matrix44`, but felt that it was better to be specific for the `c_str` method. There was also the special case for "float4" which, although I left alone, I was curious as to why we have the special cases at all (though I assume it's also an OSL thing).